lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -166,33 +166,59 @@ class DxilResourceRegisterAllocator {
       }
     }
 
-    // Allocate unallocated and bounded resources
+    // Allocate unallocated resources
     const unsigned space = AutoBindingSpace;
     typename SpacesAllocator<unsigned, T>::Allocator &alloc0 = SAlloc.Get(space);
     typename SpacesAllocator<unsigned, T>::Allocator &reservedAlloc0 = ReservedRegisters.Get(space);
     for (auto &res : resourceList) {
-      if (!res->IsAllocated() && !res->IsUnbounded()) {
-        DXASSERT(res->GetSpaceID() == 0,
-          "otherwise non-zero space has no user register assignment");
-        unsigned reg = 0;
-        if (reservedAlloc0.Find(res->GetRangeSize(), reg)) {
-          unsigned end = reg + res->GetRangeSize() - 1;
+      if (res->IsAllocated())
+        continue;
 
-          bool success = reservedAlloc0.Insert(res.get(), reg, end) == nullptr;
-          DXASSERT_NOMSG(success);
+      DXASSERT(res->GetSpaceID() == 0,
+        "otherwise non-zero space has no user register assignment");
 
-          success = alloc0.Insert(res.get(), reg, end) == nullptr;
-          DXASSERT_NOMSG(success);
+      unsigned reg = 0;
+      unsigned end = 0;
+      bool allocateSpaceFound = false;
+      if (res->IsUnbounded()) {
```

Comment:
It now looks like there's a redundant pass to try to allocate unbounded resources after this one.

Reply:
Approved, assuming the redundancy is removed as necessary.

Reply:
Oops, yes this should go away!